packages/backend/src/app.service.ts

```diff
@@ -1,8 +1,54 @@
-import { Injectable } from '@nestjs/common';
+import {BadRequestException, Injectable} from '@nestjs/common';
+import {AppDataSource} from "./config/typeorm.config";
+import {User} from "./entity/User";
+import {JwtService} from '@nestjs/jwt';
+import {TokenPayload} from "google-auth-library";
 
 @Injectable()
 export class AppService {
-  getHello(): string {
-    return 'Hello World!';
-  }
+
+    getHello(): string {
+        return 'Hello World!';
+    }
+
+    constructor(private jwtService: JwtService) {
+    }
+
+    async login(user: User) {
+        return {
+            access_token: this.jwtService.sign(
+                {
+                    id: user.id,
+                    email: user.email,
+                },
+                {
+                    secret: process.env.JWT_SECRET,
+                    expiresIn: process.env.JWT_EXPIRE_TIME,
+                }
+            ),
+        };
+    }
+
+    async signInWithGoogle(data: TokenPayload) {
+        if (!data) throw new BadRequestException();
+        let user = (await AppDataSource.manager.findBy(User, {email: data.email}))[0];
```

Comment:
doesn't it have a method e.g FindOneBy? so you have only single record in result, and no need to select [0] element with zero index?